packages/tools/generator-component/generators/app/templates/README.md

```diff
@@ -14,73 +14,108 @@
 [![Coverage Status](https://coveralls.io/repos/github/justeat/f-<%= name.default %>/badge.svg)](https://coveralls.io/github/justeat/f-<%= name.default %>)
 [![Known Vulnerabilities](https://snyk.io/test/github/justeat/f-<%= name.default %>/badge.svg?targetFile=package.json)](https://snyk.io/test/github/justeat/f-<%= name.default %>?targetFile=package.json)
 
+---
 
 ## Usage
 
-1.  Install the module using NPM or Yarn:
+### Installation
+
+Install the module using NPM or Yarn:
 
-    ```bash
-    yarn add @justeat/f-<%= name.default %>
-    ```
+```sh
+yarn add @justeat/f-<%= name.default %>
+```
 
-    ```bash
-    npm install @justeat/f-<%= name.default %>
-    ```
+```sh
+npm install @justeat/f-<%= name.default %>
+```
 
-2.  Import the component
+### Vue Applications
 
-    You can import it in your Vue SFC like this (please note that styles have to be imported separately):
+You can import it in your Vue SFC like this (please note that styles have to be imported separately):
 
-    ```
-    import <%= name.component %> from '@justeat/f-<%= name.default %>';
-    import '@justeat/f-<%= name.default %>/dist/f-<%= name.default %>.css';
+```js
+import <%= name.component %> from '@justeat/f-<%= name.default %>';
+import '@justeat/f-<%= name.default %>/dist/f-<%= name.default %>.css';
 
-    export default {
-        components: {
-            <%= name.component %>
-        }
+export default {
+    components: {
+        <%= name.component %>
     }
-    ```
+}
+```
 
-    If you are using Webpack, you can import the component dynamically to separate the `<%= name.template%>` bundle from the main `bundle.client.js`:
+If you are using Webpack, you can import the component dynamically to separate the `mega-modal` bundle from the main `bundle.client.js`:
 
-    ```
-    import '@justeat/f-<%= name.default %>/dist/f-<%= name.default %>.css';
+```js
+import '@justeat/f-<%= name.default %>/dist/f-<%= name.default %>.css';
 
-    export default {
-        components: {
-            ...
-            <%= name.component %>: () => import(/* webpackChunkName: "<%= name.template%>" */ '@justeat/f-<%= name.default %>')
-        }
+export default {
+    components: {
+        // …
+        <%= name.component %>: () => import(/* webpackChunkName: "<%= name.template%>" */ '@justeat/f-<%= name.default %>')
     }
+}
+```
+
+## Configuration
+
+### Props
+
+There may be props that allow you to customise its functionality.
+
+The props that can be defined are as follows (if any):
+
+| Prop  | Type  | Default | Description |
+| ----- | ----- | ------- | ----------- |
+
+### CSS Classes
+
+The modal has its own styles which are scoped to the component using CSS modules to prevent conflicts with existing styles on the page.
+
+In addition to this, the modal exposes some classes which you can target in your application.
```

Comment:
```suggestion
<%= name.component %> has its own styles which are scoped to the component using CSS modules to prevent conflicts with existing styles on the page.

In addition to this, <%= name.component %> exposes some classes which you can target in your application.
```